server/backend/testspg/router/test_router_application.py

```diff
@@ -237,11 +255,313 @@ def test_fam_application_endpoints_invlid_path_application_id_type(
+    access_grants_can_see_created = create_test_user_role_assignments(
+        fom_dev_access_admin_token, access_grants_able_to_see
+    )
+
+    access_grants_cannot_see_created = create_test_user_role_assignments(
+        fom_dev_access_admin_token, access_grants_not_able_to_see
+    )
+
+    # Call GET endpoint with FOM_DEV DELEGATED_ADMIN user level to
+    # obtain access grants for FOM_DEV application.
+    #
+    # Use 'COGNITO_USERNAME_IDIR_DELEGATED_ADMIN'. This user
+    # "PTOLLEST"(IDIR)" has no FOM_DEV app admin role, but has delegated
+    # admin privilege for FOM_REVIEWER and FORM_SUBMITTER_00001018, which is
+    # granted in the local sql.
+    token = jwt_utils.create_jwt_token(
+        test_rsa_key,
+        roles=[],
+        username=jwt_utils.COGNITO_USERNAME_IDIR_DELEGATED_ADMIN
+    )
+    access_grants_response = test_client_fixture.get(
+        get_application_role_assignment_end_point,
+        headers=jwt_utils.headers(token)
+    )
+    response_data = access_grants_response.json()
+
+    # --- Verify
+    # This FOM_DEV IDIR DELEGATED_ADMIN admin user with FOM_REVIEWER and
+    #   FOM_SUBMITTER_00001018 scoped roles should have visibility on user
+    #   role assignments for both IDIR and BCEID users with FOM_REVIEWER and
+    #   FOM_SUBMITTER_00001018 role.
+
+    # verify access grants created result length is the same as
+    #   access grants requested.
+    assert len(access_grants_able_to_see) == len(access_grants_can_see_created)
+    assert len(access_grants_not_able_to_see) == len(access_grants_cannot_see_created)
+
+    # verify FOM_DEV IDIR delegated admin GET result contains exact set of
+    #   user_role_xref_id from access_grants_can_see_created.
+    assert (
+        set([x["user_role_xref_id"] for x in response_data]) ==
+        set(access_grants_can_see_created)
+    )
+
+    # Verify this IDIR delegated admin can view # of records same as # records
+    #   from what was setup to be seen.
+    assert len(response_data) == len(access_grants_able_to_see)
+
+    # Verify this IDIR delegated admin has privileges to view only for users
+    #   granted with: FOM_REVIEWER, FOM_SUBMITTER_00001018
+    assert all(
+        granted["role"]["application_id"] == FOM_DEV_APPLICATION_ID and
+        (granted["role"]["role_name"] == ROLE_NAME_FOM_REVIEWER or
+         granted["role"]["role_name"] == ROLE_NAME_FOM_SUBMITTER_00001018)
+        for granted in response_data
+    )
+
+    # Verify this IDIR delegated admin can view "IDIR" users for # of records
+    #   matches what were granted with FOM_REVIEWER and FOM_SUBMITTER_00001018
+    #   role.
+    assert len([data for data in response_data if (
+        data["role"]["application_id"] == FOM_DEV_APPLICATION_ID and
+        data["user"]["user_type_code"] == UserType.IDIR and
+        (data["role"]["role_name"] == ROLE_NAME_FOM_REVIEWER or
+         data["role"]["role_name"] == ROLE_NAME_FOM_SUBMITTER_00001018)
```

Comment:
I feel we already assert the role name above, maybe here and below we only need to check the the user type code?